Looks like the test is failing 

```
Replication Status: REPLICA 
528
Tags set for http://localhost:9001/newbucket/README.md ("eb487fdb-f323-4182-876a-9b1fd77ba978").
529
Tags removed for http://localhost:9001/newbucket/README.md ("eb487fdb-f323-4182-876a-9b1fd77ba978").
530
expected tag removal to have replicated, exiting...
531
Cleaning up instances of MinIO
532
make: *** [Makefile:64: test-site-replication-ldap] Error 1
```

Not sure something needs to be fixed here.

--- a/cmd/bucket-replication.go
+++ b/cmd/bucket-replication.go
@@ -769,7 +769,7 @@ func getReplicationAction(oi1 ObjectInfo, oi2 minio.ObjectInfo, opType replicati
 	}
 
 	t, _ := tags.ParseObjectTags(oi1.UserTags)
-	if !reflect.DeepEqual(oi2.UserTags, t.ToMap()) {
+	if !reflect.DeepEqual(oi2.UserTags, t.ToMap()) || (oi2.UserTagCount != len(t.ToMap())) {
 		return replicateMetadata
 	}
 

--- a/cmd/erasure-object.go
+++ b/cmd/erasure-object.go
@@ -1573,6 +1573,7 @@ func (er erasureObjects) PutObjectTags(ctx context.Context, bucket, object strin
 	filterOnlineDisksInplace(fi, metaArr, onlineDisks)
 
 	fi.Metadata[xhttp.AmzObjectTagging] = tags
+	fi.ReplicationState = opts.PutReplicationState()
 	for k, v := range opts.UserDefined {
 		fi.Metadata[k] = v
 	}

--- a/docs/site-replication/run-multi-site-ldap.sh
+++ b/docs/site-replication/run-multi-site-ldap.sh
@@ -166,6 +166,36 @@ if [ $? -ne 0 ]; then
     exit_1;
 fi
 
+vID=$(./mc stat minio2/newbucket/README.md --json | jq .versionID)
+if [ $? -ne 0 ]; then
+    echo "expecting object to be present. exiting.."
+    exit_1;
+fi
+./mc tag set --version-id "${vID}" minio2/newbucket/README.md "k=v"
+if [ $? -ne 0 ]; then
+    echo "expecting tag set to be successful. exiting.."
+    exit_1;
+fi
+sleep 5
+
+./mc tag remove --version-id "${vID}" minio2/newbucket/README.md
+if [ $? -ne 0 ]; then
+    echo "expecting tag removal to be successful. exiting.."
+    exit_1;
+fi
+sleep 5
+
+replStatus_minio2=$(./mc stat minio2/newbucket/README.md --json | jq -r .replicationStatus)
+if [ $? -ne 0 ]; then
+    echo "expecting object to be present. exiting.."
+    exit_1;
+fi
+
+if [ ${replStatus_minio2} != "COMPLETED" ]; then
+    echo "expected tag removal to have replicated, exiting..."
+    exit_1;
+fi
+
 ./mc rm minio3/newbucket/README.md
 sleep 5
 

--- a/docs/site-replication/run-multi-site-minio-idp.sh
+++ b/docs/site-replication/run-multi-site-minio-idp.sh
@@ -158,6 +158,36 @@ if [ $? -ne 0 ]; then
     exit_1;
 fi
 
+vID=$(./mc stat minio2/newbucket/README.md --json | jq .versionID)
+if [ $? -ne 0 ]; then
+    echo "expecting object to be present. exiting.."
+    exit_1;
+fi
+./mc tag set --version-id "${vID}" minio2/newbucket/README.md "k=v"
+if [ $? -ne 0 ]; then
+    echo "expecting tag set to be successful. exiting.."
+    exit_1;
+fi
+sleep 5
+
+./mc tag remove --version-id "${vID}" minio2/newbucket/README.md
+if [ $? -ne 0 ]; then
+    echo "expecting tag removal to be successful. exiting.."
+    exit_1;
+fi
+sleep 5
+
+replStatus_minio2=$(./mc stat minio2/newbucket/README.md --json | jq -r .replicationStatus )
+if [ $? -ne 0 ]; then
+    echo "expecting object to be present. exiting.."
+    exit_1;
+fi
+
+if [ ${replStatus_minio2} != "COMPLETED" ]; then
+    echo "expected tag removal to have replicated, exiting..."
+    exit_1;
+fi
+
 ./mc rm minio3/newbucket/README.md
 sleep 5
 

